builtin/providers/aws/resource_aws_sns_topic_subscription.go

```diff
@@ -192,6 +218,76 @@ func subscribeToSNSTopic(d *schema.ResourceData, snsconn *sns.SNS) (output *sns.
 		return nil, fmt.Errorf("Error creating SNS topic: %s", err)
 	}
 
-	log.Printf("[DEBUG] Created new subscription!")
+	log.Printf("[DEBUG] Finished subscribing to topic %s with subscription arn %s", topic_arn, *output.SubscriptionArn)
+
+	if strings.Contains(protocol, "http") && subscriptionHasPendingConfirmation(output.SubscriptionArn) {
+
+		log.Printf("[DEBUG] SNS create topic subscritpion is pending so fetching the subscription list for topic : %s (%s) @ '%s'", endpoint, protocol, topic_arn)
+
+		for i := 0; i < max_fetch_retries && subscriptionHasPendingConfirmation(output.SubscriptionArn); i++ {
```

Comment:
We have a nice helper for this which also uses exponential back-off, see `resource.Retry`, here's an example: https://github.com/hashicorp/terraform/blob/a9803d646c2f578c69f5b825971abc1a66cde0a1/builtin/providers/aws/resource_aws_ecs_cluster.go#L88-L114

Then `max_fetch_retries` & `fetch_retry_delay` would turn into `confirmation_timeout_in_minutes`. I think having 1min as default timeout is sensible settings, in most cases it will be confirmed much sooner anyway, I assume.

Let me know if you need any help implementing that helper.